create-package/#19

InderdeepSingh355 opened the pull request:
- Added DSM demo case in package structure
- GitHub workflow for publishing package to TestPyPI and PyPI
- Added import package instructions to readme
- Adjusted Jupyter notebook to import package vlinder

MikeVerdaasdonk297 commented:
Does not succeed in pytest (python3 -m pytest). See error below.
<img width="1792" alt="Scherm­afbeelding 2024-01-19 om 15 55 39" src="https://github.com/responsible-business-decision-making/trbs/assets/140519472/bf938021-9680-4250-b1d2-0b02fcc92f79">

MikeVerdaasdonk297 commented:
DSM can not be used because it is not in the vlinder package yet.
![Scherm­afbeelding 2024-01-19 om 16 42 04 (2)](https://github.com/responsible-business-decision-making/trbs/assets/140519472/0e22d390-1a46-440c-a068-1cb70a2d115b)

InderdeepSingh355 commented:
- Changed test scripts according to vlinder package. Added vlinder in import statements and data path. Now test scripts work.
- Added DSM case to vlinder package version 0.1.0 to PyPI.

tina-tami commented on Analyze case with vlinder.ipynb:
```diff
@@ -5,23 +5,42 @@
+    "!pip install openpyxl\n",
+    "!pip install matplotlib\n",
+    "!pip install xlsxwriter\n",
+    "!pip install vlinder --upgrade"
```
waarom hier upgrade en bij andere packages niet? kunnen we dit omzeilen?

InderdeepSingh355 replied:
Omdat bij vlinder pakt hij niet de latest versie zonder --upgrade, als je al een oudere versie van vlinder hebt geïnstalleerd. Met --upgrade pakt hij wel altijd de latest versie van vlinder.

tina-tami commented on Analyze case with vlinder.ipynb:
```diff
@@ -638,31 +519,15 @@
    "source": [
     "output_path = Path.cwd() / 'data' / name \n",
     "case.transform(output_path, \"json\")\n",
     "# case.transform(output_path, \"xlsx\")\n",
```
waarom deze uitgecomment?

InderdeepSingh355 replied:
Deze was al uitgecomment in de oorspronkelijke analyze_case.ipynb notebook, ik denk omdat bij het inlezen al een xlsx format is gebruikt (bij case = TheResponsibleBusinessSimulator(path, file_format, name)).

tina-tami reviewed (changes requested):
passed unit tests + pep8, few questions
